src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -3616,14 +3604,19 @@ export class StudyViewPageStore {
         );
 
         _.reduce(
-            this.survivalPlots,
+            this.survivalPlots.result,
             (acc: { [id: string]: ChartMeta }, survivalPlot) => {
                 acc[survivalPlot.id] = {
                     uniqueKey: survivalPlot.id,
                     dataType: getChartMetaDataType(survivalPlot.id),
                     patientAttribute: true,
                     displayName: survivalPlot.title,
-                    priority: getDefaultPriorityByUniqueKey(survivalPlot.id),
+                    // use survival status attribute's priority as KM plot's priority for non-reserved plots
+                    priority:
+                        STUDY_VIEW_CONFIG.priority[survivalPlot.id] ||
+                        getPriorityByClinicalAttribute(
+                            survivalPlot.survivalStatusAttribute
+                        ),
```

Comment:
Derive priority from `STUDY_VIEW_CONFIG` or related clinical attribute.

Reply:
i'm surprised that config takes precedence here (over data). is that right?

Reply:
Yes, it is. For example, in this file: https://github.com/cBioPortal/cbioportal-frontend/blob/master/src/config/serverConfigDefaults.ts#L122-L125. We have priority four types of `KM plot`. We want to show them in front of the other charts. We can just ignore the priority in the related `survival status attribute` (e.g. `OS_STATUS`).